### PR TITLE
Suggestion how to use the custom field class

### PR DESCRIPTION
**Pull request by sandstromviktor**
You added the `spinner` bool to the `get_common_field` function. I think it is better to add it to the `CustomField` class, and then let that `spinner` decide in the template if it should add or not add the spinner animation. 

This does not require two separate templates and, i think, makes the logic a bit cleaner.

### CODE REVIEW

File: apps/forms/base.py [hunk 1]
```diff
@@ -124,27 +124,22 @@ def get_common_field(self, field_name: str, spinner: bool = False, **kwargs):
...
+
+        spinner = kwargs.pop("spinner", False)
+
+        template = "apps/custom_field.html"
+        base_args = dict(
+            css_class="form-control form-control-with-spinner" if spinner else "form-control",
+            wrapper_class="mb-3",
+            rows=3,
+            help_message=HELP_MESSAGE_MAP.get(field_name, ""),
+            spinner=spinner,
```
Comment (sandstromviktor) [inline apps/forms/base.py:136]: this adds spinner to the `CustomField` class

File: apps/forms/base.py [hunk 2]
```diff
@@ -124,27 +124,22 @@ def get_common_field(self, field_name: str, spinner: bool = False, **kwargs):
...
+        )
 
         base_args.update(kwargs)
         field = CustomField(field_name, **base_args)
         field.set_template(template)
-        return Div(field)
+        return Div(field, css_class="form-input-with-spinner" if spinner else None)
```
Comment (sandstromviktor) [inline apps/forms/base.py:142]: No need to double wrap the field in a Div, so just add the `form-input-with-spinner` css class here if `spinner` is `True`

File: apps/forms/custom.py [hunk 3]
```diff
@@ -28,11 +28,8 @@ def _setup_form_helper(self):
         body = Div(
             self.get_common_field("name", placeholder="Name your app"),
             self.get_common_field("description", rows=3),
-            Div(
-                self.get_common_field(
-                    "subdomain", placeholder="Enter a subdomain or leave blank for a random one", spinner=True
-                ),
-                css_class="form-input-with-spinner",
+            self.get_common_field(
+                "subdomain", placeholder="Enter a subdomain or leave blank for a random one", spinner=True
```
Comment (sandstromviktor) [inline apps/forms/custom.py:32]: The above makes this cleaner.

File: templates/apps/custom_field.html [hunk 4]
```diff
@@ -1,12 +1,25 @@
...
+    </div>
+
+    <div style="display: block; clear:both;">
+        {% if spinner %}
+        <div style="float:right;">
+            <div id="formInputSpinner" class="spinner-border text-primary" style="display: none;" role="status">
+                <span class="sr-only"></span>
+              </div>
+        </div>
+        {% endif %}
```
Comment (sandstromviktor) [inline templates/apps/custom_field.html:15]: Adding the spinner here.

**Review by alfredeen — APPROVED** [review]
Nice suggestions! LGTM

**Pull request merged**